pennylane/ops/qubit.py

```diff
@@ -2547,6 +2547,197 @@ def diagonalizing_gates(self):
+            [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1],
+            [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0],
+            [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0],
+            [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0],
+        ]
+    )
+
+    @classmethod
+    def _matrix(cls, *params):
+        return QubitCarry.matrix
+
+    @staticmethod
+    def decomposition(wires):
+        decomp_ops = [
+            qml.Toffoli(wires=wires[1:]),
+            qml.CNOT(wires=[wires[1], wires[2]]),
+            qml.Toffoli(wires=[wires[0], wires[2], wires[3]]),
+        ]
+        return decomp_ops
+
+
+class QubitSum(Operation):
+    r"""QubitSum(wires)
+    Apply a ``QubitSum`` operation on three input wires.
+
+    This operation performs the transformation:
+
+    .. math::
+        |a\rangle |b\rangle |c\rangle \rightarrow |a\rangle |b\rangle |a\oplus b\oplus c\rangle
+
+
+    .. figure:: ../../_static/ops/QubitSum.svg
+        :align: center
+        :width: 60%
+        :target: javascript:void(0);
+
+    See `here <https://arxiv.org/abs/quant-ph/0008033v1>`__ for more information.
+
+    **Details:**
+
+    * Number of wires: 3
+    * Number of parameters: 0
+
+    Args:
+        wires (Sequence[int]): the wires the operation acts on
+
+    **Example**
+
+    The ``QubitSum`` operation maps the state :math:`|010\rangle` to :math:`|011\rangle`, with the
+    final wire holding the modulo-two sum of the first two wires:
+
+    .. code-block::
+
+        input_bitstring = (0, 1, 0)
+
+        @qml.qnode(dev)
+        def circuit(basis_state):
+            qml.BasisState(basis_state, wires = [0, 1, 2])
+            qml.QubitSum(wires=[0, 1, 2])
+            return qml.probs(wires=[0, 1, 2])
+
+        probs = circuit(input_bitstring)
+        probs_indx = np.argwhere(probs == 1).flatten()[0]
+        bitstrings = list(itertools.product(range(2), repeat=3))
+        output_bitstring = bitstrings[probs_indx]
+
+    The output bitstring is
+
+    >>> output_bitstring
+    (0, 1, 1)
+
+    The action of ``QubitSum`` is to add wires ``0``, ``1``, and ``2``. The modulo-two result is
+    output in wire ``2``. In this case, :math:`0 \oplus 1 \oplus 0 = 1`, so we have:
+
+    >>> abc_sum = output_bitstring[2]
+    >>> abc_sum
+    1
+    """
+    num_params = 0
+    num_wires = 3
+    par_domain = None
+    matrix = np.array(
+        [
+            [1, 0, 0, 0, 0, 0, 0, 0],
+            [0, 1, 0, 0, 0, 0, 0, 0],
+            [0, 0, 0, 1, 0, 0, 0, 0],
+            [0, 0, 1, 0, 0, 0, 0, 0],
+            [0, 0, 0, 0, 0, 1, 0, 0],
+            [0, 0, 0, 0, 1, 0, 0, 0],
+            [0, 0, 0, 0, 0, 0, 1, 0],
+            [0, 0, 0, 0, 0, 0, 0, 1],
+        ]
+    )
+
+    @classmethod
+    def _matrix(cls, *params):
+        return QubitSum.matrix
+
+    @staticmethod
+    def decomposition(wires):
+        decomp_ops = [qml.CNOT(wires=[wires[1], wires[2]]), qml.CNOT(wires=[wires[0], wires[2]])]
+        return decomp_ops
+
+
```

Comment:
Since `QubitSum` is it's own adjoint, you should 
```suggestion

def adjoint(self):
    return QubitSum(wires=self.wires)
```